### ISSUE TITLE
Python 2.6: false positive W605 invalid escape sequence

### ISSUE THREAD

**Issue by jwilk** [PyCQA/pycodestyle#755]
I'm getting this false positive:
```console
$ echo 'br"\s"' | python2.6 -m pycodestyle /dev/stdin
/dev/stdin:1:3: W605 invalid escape sequence '\s'
```
I suspect this is caused by [Python issue #15054](https://bugs.python.org/issue15054).

You may want to disable W605 for Python 2.6.

**Comment by sigmavirus24** [comment]
I don't think our checks are specific to Python versions. That said, I think the error message could be improved. And a PR to clarify what's "invalid" about it (for Python 3.7) is needed. That would allow folks to `--ignore W605` if they never plan to use Python 3.7

**Comment by seeeturtle** [comment]
@sigmavirus24, afaik, the warning is changed since python 3.6
(Note https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals)
isn't it?

**Comment by sigmavirus24** [comment]
I'm sorry @seeeturtle, could you rephrase? I'm having trouble understanding your question.

**Comment by seeeturtle** [comment]
> Changed in version 3.6: Unrecognized escape sequences produce a DeprecationWarning. In some future version of Python they will be a SyntaxError.

https://bugs.python.org/issue27364

Based on the above two, I thought that the patch of rasing `DeprecationWarning` for invalid escape sequence is changed since python 3.6.
However, your comment seems to mean to patch only python under 3.7.

Is this right?

**Comment by seeeturtle** [comment]
No problem :)
Btw, maybe I can send PR for modifying docs about w605 to show that it will not gonna produce error until python 3.7.

**Comment by sigmavirus24** [comment]
@seeeturtle absolutely!

**Comment by gdevenyi** [comment]
This also triggers a false positive for lines defining a regex:

```python
import re
re.match("^([-+]?\d+|^\d+%)$", string)
```
```
W605 invalid escape sequence '\d'
```

**Comment by FichteFoll** [comment]
@gdevenyi this is intended as you should be using a raw strings for regular expressions in most cases. `\d` *is* an invalid string escape sequence.

**Comment by serhiy-storchaka** [comment]
`br"\s"` doesn't contain an invalid escape sequence because it is a *raw* string.

The sequence `\s` is considered invalid only in *non-raw* strings.
